test/support/resources.ex

```diff
@@ -20,6 +20,23 @@ defmodule LiveAdminTest.User do
   def failing_action(_, _), do: {:error, "failed"}
 end
 
+defmodule LiveAdminTest.CustomStringType do
+  use Ecto.Type
+  @behaviour LiveAdmin.Type
+
+  # Ecto.Type callbacks
+  def type, do: :string
+
+  def cast(value) when is_binary(value), do: {:ok, String.trim(value)}
+  def cast(_), do: :error
+
+  def load(value), do: {:ok, value}
+  def dump(value), do: {:ok, value}
+
+  # LiveAdmin.Type callbacks
+  def render_as(), do: :string
```

Comment:
@tfwright no, my understanding is that's the current behavior. The idea is, if your custom type can be treated like a primitive type, you add this line and LiveAdmin handles it as if it were that type. Alternatively you could attempt to read the type from line 28 above and not require conforming to a new behavior but this is more explicit. This interface could also be extended to allow for custom types to ovveride more default behavior but this is enough for mine and probably many use cases.

Reply:
> Alternatively, you could try to read the type from line 28 above and not require conforming to a new behavior but this is more explicit and doesn't change the behavior of any existing liveadmin dasboard.

Despite concerns about affecting current behavior, I think this is the better way to go vs creating a new api, especially since there is already a `render_with` this would easily be confused with.